core/src/main/java/hudson/tasks/BatchFile.java

```diff
@@ -72,8 +93,28 @@ public String getDisplayName() {
         }
 
         @Override
-        public Builder newInstance(StaplerRequest req, JSONObject data) {
-            return new BatchFile(data.getString("command"));
+        public Builder newInstance(StaplerRequest req, JSONObject formData) {
+            return req.bindJSON(BatchFile.class,formData);
```

Comment:
Of course [this annotation could soon be obsolete](https://github.com/jenkinsci/jenkins/pull/2519/files#diff-d2f501ee1607eedcf2c217330f500c0eR566).